File: leetcode/editor/cn/292_Nim 游戏_test.java
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.BDDAssertions.then;


class NimGameTest {

    final Solution solution = new Solution();

    @Test
    void givenNormal_when_thenSuccess() {
        then(solution.canWinNim(1)).isTrue();
        then(solution.canWinNim(4)).isFalse();
    }

    //leetcode submit region begin(Prohibit modification and deletion)
    class Solution {

        public boolean canWinNim(int n) {
            return (n % 4) != 0;
        }

    }
//leetcode submit region end(Prohibit modification and deletion)


}

File: leetcode/editor/cn/292_Nim 游戏_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when292_thenSuccess(t *testing.T) {
	assert.Equal(t, true, canWinNim(1), "shouldEqual")
	assert.Equal(t, false, canWinNim(4), "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
func canWinNim(n int) bool {
	return (n % 4) != 0
}

//leetcode submit region end(Prohibit modification and deletion)
